src/app/charts/core/files/secret/tpl/arkcase-server.yaml

```diff
@@ -75,8 +75,17 @@ fileManagement:
 {{- end }}
 
 {{- if (eq "box" $content.dialect) }}
-box:
-  privateKeyPath: "${CONTENT_MAIN_PRIVATE_KEY_PATH:/app/home/.content-private-key}"
+  {{- $cmInfo := (.Files.Get "cminfo.yaml" | fromYaml) }}
+  {{- $privateKeyDict := dict "path" "/app/home/.content-private-key" }}
+
+  {{- range $key, $cm := $cmInfo }}
+    {{- if and (hasKey $cm "private-key") (hasKey (get $cm "private-key") "path") }}
+      {{- $_ := set $privateKeyDict "path" (get (get $cm "private-key") "path") }}
+    {{- end }}
+  {{- end }}
```

Comment:
There's a simpler way to do this: look for [dig](https://helm.sh/docs/chart_template_guide/function_list/#dig) ;)

Here's a one-liner to do the same thing, given the fact that you're expecting a predictable structure:

{{- $privateKeyPath := dig "private-key" "path" "/app/home/.content-private-key" $cmInfo -}}

Also, [this doc is your friend](https://masterminds.github.io/sprig/) ;)